topdown/query.go

```diff
@@ -236,6 +237,14 @@ func (q *Query) WithStrictBuiltinErrors(yes bool) *Query {
 	return q
 }
 
+// WithBuiltinErrors tells the evaluator to pass built-in function errors to provided function.
+// It presents an alterative to using `WithStrictBuiltinErrors(true)`: the errors that happen
+// can be inspected, but don't affect the evaluation.
+func (q *Query) WithBuiltinErrors(f func(errs ...error)) *Query {
```

Comment:
Just wondering about the performance implications of this API (the closure, the variadic arguments, etc.) Do we incur extra heap allocations because of the style of the API? The other thing that jumps to mind is concurrency; if we ever make topdown parallelize query execution, we'll need to be careful to ensure that this function is not invoked concurrently. That seems easy enough to solve for though.